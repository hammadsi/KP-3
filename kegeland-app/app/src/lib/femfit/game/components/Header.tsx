import React from 'react';
import {Image, StyleSheet, View} from 'react-native';
import {TouchableOpacity} from 'react-native-gesture-handler';
import {Text, useTheme} from 'react-native-paper';

import CoinSprite from '~assets/femfit/gold.png';
import Icon from '~components/Icon';

import constants from '../constants';
const {HEADER_HEIGHT, COIN_SIZE} = constants;

export type HeaderProps = {
  score: number;
  toggle: () => void;
};

/**
 * Component for rendering the game header panel
 * @param param0 the props
 * @see {@link HeaderProps}
 */
const Header: React.FC<HeaderProps> = ({score, toggle}) => {
  const {colors} = useTheme();
  return (
    <View
      style={[
        styles.header,
        {height: HEADER_HEIGHT, backgroundColor: colors.surface},
      ]}>
      <View style={styles.scoreContainer}>
        <Text style={styles.score}>{score}</Text>
        <Image
          style={styles.scoreSprite}
          source={CoinSprite}
          resizeMode="stretch"
        />
      </View>
      <View>
        <TouchableOpacity onPress={toggle}>
          <Icon icon="menu" color={colors.primary} size={28} />
        </TouchableOpacity>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  header: {
    flexDirection: 'row',
    zIndex: 1000,
    justifyContent: 'space-between',
    alignItems: 'center',
    paddingHorizontal: 8,
  },
  scoreContainer: {
    flexDirection: 'row',
    alignItems: 'center',
  },
  score: {
    fontSize: 20,
    fontWeight: 'bold',
  },
  scoreSprite: {
    marginLeft: 8,
    width: COIN_SIZE,
    height: COIN_SIZE,
  },
});

export default Header;
