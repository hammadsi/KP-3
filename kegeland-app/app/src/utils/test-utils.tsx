/* istanbul ignore file */
import React from 'react';
import {NavigationContainer} from '@react-navigation/native';
import {Provider as PaperProvider} from 'react-native-paper';
import {render, RenderOptions} from '@testing-library/react-native';
import {Provider as ReduxProvider} from 'react-redux';

import {store} from '~state/store';
import {lightTheme} from '~constants/theme';
import {initialStore, mockStore} from '~state/ducks/__mocks__/store.mock';

type TestWrapperProps = {
  children: React.ReactNode;
  store?: typeof store;
};

/**
 * Wrapper component with redux state. Used when testing components
 * which requires redux.
 * @param param0 test wrapper props @see {@link TestWrapperProps}
 * @returns
 */
export const TestWrapper: React.FC<TestWrapperProps> = ({
  children,
  ...props
}) => {
  return (
    <ReduxProvider store={props.store || mockStore(initialStore)}>
      <PaperProvider theme={lightTheme}>
        <NavigationContainer theme={lightTheme}>{children}</NavigationContainer>
      </PaperProvider>
    </ReduxProvider>
  );
};

/**
 * Custom test renderer which wraps components with a redux store provider
 * @param ui the component to render
 * @param options renderer options
 * @returns test rendered componnet
 */
export const customRender = <T extends any>(
  ui: React.ReactElement<T>,
  options?: {
    wrapperProps?: {
      store: any;
    };
    renderOptions?: RenderOptions;
  },
) =>
  render(ui, {
    wrapper: (props) => <TestWrapper {...props} {...options?.wrapperProps} />,
    ...options?.renderOptions,
  });

// re-export everything
export * from '@testing-library/react-native';

// override render method
export {customRender as render};
