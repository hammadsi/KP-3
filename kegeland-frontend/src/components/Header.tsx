import { Box, Flex, Icon, useMediaQuery, Text, HStack } from '@chakra-ui/react';
import { MdMenu, MdChevronLeft } from 'react-icons/md';
import { useNavigate } from 'react-router-dom';

import useBreadcrumbs from '../hooks/useBreadcrumbs';

import Breadcrumbs from './Breadcrumbs';

type HeaderProps = {
  isSidebarOpen: boolean;
  toggleSidebar: () => void;
};

const Header: React.FC<HeaderProps> = ({ toggleSidebar, isSidebarOpen }) => {
  const navigate = useNavigate();
  const { crumbs } = useBreadcrumbs();
  const [isGreaterThanLg] = useMediaQuery('(min-width: 62em)');
  const moveBtn = isSidebarOpen && !isGreaterThanLg;
  return (
    <Box
      display="flex"
      paddingX={4}
      width={'100%'}
      height={'60px'}
      flexDirection="row"
      alignItems="center"
      bgColor="white"
      boxShadow="md"
      zIndex={1000}>
      <Flex
        flexDir="row"
        zIndex={999}
        overflow="hidden"
        marginX="auto"
        w="100%"
        maxW="8xl"
        justify="space-between"
        align="stretch">
        <Flex justify="center" align="center">
          <Icon
            as={MdMenu}
            onClick={toggleSidebar}
            position={moveBtn ? 'absolute' : 'relative'}
            left={moveBtn ? '300px' : 0}
            color="primary.600"
            transform="all 300ms"
            _hover={{ cursor: 'pointer', color: 'primary.700' }}
            aria-label="Toggle sidebar"
            width="40px"
            height="40px"
          />
          {isGreaterThanLg && crumbs.length > 0 && (
            <Breadcrumbs marginLeft={4} crumbs={crumbs} />
          )}
        </Flex>
        <Flex justify="end" flex={1} flexGrow={1}>
          {crumbs.length > 0 && (
            <HStack
              spacing={0}
              onClick={() => navigate(-1)}
              color="primary.600"
              transform="color 300ms"
              _hover={{ cursor: 'pointer', color: 'primary.700' }}>
              <Icon
                as={MdChevronLeft}
                aria-label="Toggle sidebar"
                width="24px"
                height="24px"
              />
              <Text fontWeight="semibold">Go back</Text>
            </HStack>
          )}
        </Flex>
      </Flex>
    </Box>
  );
};

export default Header;
