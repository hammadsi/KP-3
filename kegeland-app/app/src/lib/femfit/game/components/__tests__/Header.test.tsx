import React from 'react';

import {render} from '~utils/test-utils';

import Header, {HeaderProps} from '../Header';

describe('Test Header-component', () => {
  it('should render correctly', () => {
    const props: HeaderProps = {
      score: 10,
      toggle: jest.fn(),
    };
    const component = <Header {...props} />;
    const tree = render(component);
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
