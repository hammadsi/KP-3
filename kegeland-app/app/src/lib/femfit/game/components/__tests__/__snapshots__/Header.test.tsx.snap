// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Header-component should render correctly 1`] = `
<View
  collapsable={false}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingHorizontal": 8,
          "zIndex": 1000,
        },
        Object {
          "backgroundColor": "#ffffff",
          "height": 50,
        },
      ]
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
            },
            Object {
              "textAlign": "left",
            },
            Object {
              "fontSize": 20,
              "fontWeight": "bold",
            },
          ]
        }
      >
        10
      </Text>
      <Image
        resizeMode="stretch"
        source={
          Object {
            "testUri": "../../../assets/femfit/gold.png",
          }
        }
        style={
          Object {
            "height": 25,
            "marginLeft": 8,
            "width": 25,
          }
        }
      />
    </View>
    <View>
      <RNGestureHandlerButton
        collapsable={false}
        delayLongPress={600}
        enabled={true}
        exclusive={true}
        handlerTag={1}
        handlerType="NativeViewGestureHandler"
        onGestureEvent={[Function]}
        onGestureHandlerEvent={[Function]}
        onGestureHandlerStateChange={[Function]}
        onHandlerStateChange={[Function]}
        rippleColor={0}
        touchSoundDisabled={false}
      >
        <View
          accessible={true}
          collapsable={false}
          nativeID="animatedComponent"
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              Array [
                Object {
                  "color": "#00b0ff",
                  "fontSize": 28,
                },
                Array [
                  Object {
                    "alignSelf": "center",
                  },
                  undefined,
                ],
                Object {
                  "fontFamily": "Material Design Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            Û∞çú
          </Text>
        </View>
      </RNGestureHandlerButton>
    </View>
  </View>
</View>
`;
